### PR TITLE
Curly quotes!

### PR DESCRIPTION
**Pull request by mdznr**
Instead of using straight quotes, properly use the curly ones. Thanks!

### CODE REVIEW

**Comment by swisspol** [comment]
Why would you want that?

No matter what, only UI facing quotes should be replaced (nib files and localized strings), not anything else like `XLOG_VERBOSE(...)` and friends.

Finally I'm skeptical such a change would be maintained i.e. when adding new strings, people won't remember to use curly quotes.

**Comment by mdznr** [comment]
1. Curly quotes are typographically more correct. Open an app from Apple and you’ll see this is how it appears. Many apps using AppKit automatically insert the correct curly quote mark when typing. Open up Pages.app and you’ll see this is the behaviour.

2. I’ve removed the curly quotes from logging. This pull request shall be for user-facing strings.

3. Perhaps. I’ll have to open new pull requests to fix those if/when that happens.

**Comment by swisspol** [comment]
GitUp is a developer app though, not a word processor or presentation making app. In this context, ASCII are pretty much always used (when was the last time you saw curly quotes in Terminal?).

If a ton of people want this, I don't mind merging it though :smile:

**Comment by mdznr** [comment]
Right, but these are user interface strings presented to users in a Mac app, not output from the command line.

This app helps to keep me from using the Terminal for things like git 😝

**Comment by salutis** [comment]
+1 for merging. Proper typography in the UI is important.

**Comment by douglashill** [comment]
I am also in support of this. GitUp should aim to do better than Terminal. @swisspol, you’ve made Git much easier to use: do you have no ambition for GitUp to be used outside of software development?

**Comment by swisspol** [comment]
@douglashill I can't even :smile:

File: Application/AppDelegate.m [hunk 1]
```diff
@@ -439,7 +439,7 @@ - (NSDictionary*)_processToolCommand:(NSDictionary*)input {
   } else if ([command isEqualToString:@kToolCommand_Stash]) {
     [self _openRepositoryWithURL:[NSURL fileURLWithPath:repository] withCloneMode:kCloneMode_None windowModeID:kWindowModeID_Stashes];
   } else {
-    return @{kToolDictionaryKey_Error: [NSString stringWithFormat:@"Unknown command '%@'", command]};
+    return @{kToolDictionaryKey_Error: [NSString stringWithFormat:@"Unknown command ‘%@’", command]};
```
Comment (swisspol) [inline Application/AppDelegate.m]: This is not a UI string, it's passed on to the tool for display in Terminal.

File: Core/GCReflogMessages.h [hunk 2]
```diff
@@ -36,10 +36,10 @@
 #define kGCReflogMessageFormat_Git_Branch_Renamed @kGCReflogMessagePrefix_Git_Branch_Renamed "%@ to %@"
 
 #define kGCReflogMessagePrefix_Git_Revert "revert: "
-#define kGCReflogMessageFormat_Git_Revert @kGCReflogMessagePrefix_Git_Revert "Revert \"%@\""
```
Comment (swisspol) [inline Core/GCReflogMessages.h]: Strings in these file must not be modified as they go into the reflog.

File: Core/GCRemote.m [hunk 3]
```diff
@@ -183,7 +183,7 @@ - (NSUInteger)_transfer:(git_direction)direction withRemote:(git_remote*)remote
   if (status != GIT_OK) {
     LOG_LIBGIT2_ERROR(status);
     if (error) {
-      *error = GCNewError(status, [NSString stringWithFormat:@"Failed connecting to \"%s\" remote: %@", git_remote_name(remote), GetLastGitErrorMessage()]);  // We can't use CALL_LIBGIT2_FUNCTION_GOTO() as we need to customize the error message
```
Comment (swisspol) [inline Core/GCRemote.m]: Low-level error messages should not be modified.

File: Core/GCRemote.m [hunk 4]
```diff
@@ -364,7 +364,7 @@ - (BOOL)fetchRemoteBranch:(GCRemoteBranch*)branch tagMode:(GCFetchTagMode)mode u
       goto cleanup;  // TODO: What if there is more than one match?
     }
   }
-  GC_SET_GENERIC_ERROR(@"No matching refspec for \"%@\"", branch.name);
```
Comment (swisspol) [inline Core/GCRemote.m]: Same issue.

File: Extensions/GCRepository+Utilities.m [hunk 5]
```diff
@@ -142,7 +142,7 @@ - (BOOL)moveFileFromPath:(NSString*)fromPath toPath:(NSString*)toPath force:(BOO
   }
   
   if (![[NSFileManager defaultManager] fileExistsAtPath:sourcePath isDirectory:&isDirectory] || isDirectory) {
-    GC_SET_GENERIC_ERROR(@"No file at \"%@\"", sourcePath);
```
Comment (swisspol) [inline Extensions/GCRepository+Utilities.m]: Don't modify.

File: Views/GICommitRewriterViewController.m [hunk 6]
```diff
@@ -159,7 +159,7 @@ - (BOOL)startRewritingCommit:(GCHistoryCommit*)commit error:(NSError**)error {
   [self.repository notifyRepositoryChanged];
   
   _targetCommit = commit;
-  _titleTextField.stringValue = [NSString stringWithFormat:@"\"%@\" <%@>", _targetCommit.summary, _targetCommit.shortSHA1];
```
Comment (swisspol) [inline Views/GICommitRewriterViewController.m]: This should be changed into NSLocalizedString() in a separate and prior commit to make it clear it's a UI string.

File: Views/GICommitSplitterViewController.m [hunk 7]
```diff
@@ -87,7 +87,7 @@ - (void)viewWillShow {
   XLOG_DEBUG_CHECK(_commit);
   [super viewWillShow];
   
-  _titleTextField.stringValue = [NSString stringWithFormat:@"\"%@\" <%@>", _commit.summary, _commit.shortSHA1];
```
Comment (swisspol) [inline Views/GICommitSplitterViewController.m]: This should be changed into NSLocalizedString() in a separate and prior commit to make it clear it's a UI string.

File: Views/GIConflictResolverViewController.m [hunk 8]
```diff
@@ -61,8 +61,8 @@ - (void)viewWillShow {
   XLOG_DEBUG_CHECK(self.repository.statusMode == kGCLiveRepositoryStatusMode_Disabled);
   self.repository.statusMode = kGCLiveRepositoryStatusMode_Unified;
   
-  _oursTextField.stringValue = [NSString stringWithFormat:@"\"%@\" <%@>", _ourCommit.summary, _ourCommit.shortSHA1];
```
Comment (swisspol) [inline Views/GIConflictResolverViewController.m]: This should be changed into NSLocalizedString() in a separate and prior commit to make it clear it's a UI string.

File: Views/GIConflictResolverViewController.m [hunk 9]
```diff
@@ -61,8 +61,8 @@ - (void)viewWillShow {
...
-  _theirsTextField.stringValue = [NSString stringWithFormat:@"\"%@\" <%@>", _theirCommit.summary, _theirCommit.shortSHA1];
```
Comment (swisspol) [inline Views/GIConflictResolverViewController.m]: Same issue.

File: Views/GIDiffViewController.m [hunk 10]
```diff
@@ -79,8 +79,8 @@ - (void)setCommit:(GCCommit*)commit withParentCommit:(GCCommit*)parentCommit {
       [_diffContentsViewController setDeltas:diff.deltas usingConflicts:nil];
       [_diffFilesViewController setDeltas:diff.deltas usingConflicts:nil];
       
-      _fromTextField.stringValue = [NSString stringWithFormat:@"\"%@\" <%@> (%@)", _commit.summary, _commit.shortSHA1, [_dateFormatter stringFromDate:_commit.date]];
```
Comment (swisspol) [inline Views/GIDiffViewController.m]: This should be changed into NSLocalizedString() in a separate and prior commit to make it clear it's a UI string.

File: Views/GIDiffViewController.m [hunk 11]
```diff
@@ -79,8 +79,8 @@ - (void)setCommit:(GCCommit*)commit withParentCommit:(GCCommit*)parentCommit {
...
-      _toTextField.stringValue = [NSString stringWithFormat:@"\"%@\" <%@> (%@)", _parentCommit.summary, _parentCommit.shortSHA1, [_dateFormatter stringFromDate:_parentCommit.date]];
```
Comment (swisspol) [inline Views/GIDiffViewController.m]: Same issue.

**Comment by swisspol** [comment]
It seems the only strings that should be modified are the ones inside `NSLocalizedString()`, inside XIBs or inside Help.plist.

Once this is confirmed, the next step would be to add a shell script phase that runs in Release / Distribution configurations and check for use of regular quotes in these locations. This would automatically prevent any regression.

**Comment by mdznr** [comment]
Thanks for the feedback! Will work on it tonight…

File: Views/GICommitSplitterViewController.m (hunk 7 above)
Comment (mdznr) [inline Views/GICommitSplitterViewController.m]: I'll use [NSString localizedStringWithFormat:].

File: Views/GICommitRewriterViewController.m (hunk 6 above)
Comment (mdznr) [inline Views/GICommitRewriterViewController.m]: I'll use [NSString localizedStringWithFormat:].

File: Views/GIConflictResolverViewController.m (hunk 8 above)
Comment (mdznr) [inline Views/GIConflictResolverViewController.m]: I'll use `[NSString localizedStringWithFormat:]`.

File: Views/GICommitRewriterViewController.m (hunk 6 above)
Comment (swisspol) [inline Views/GICommitRewriterViewController.m]: For consistency with the rest of the source code, you actually need to use NSLocalizedString().

**Comment by jirutka** [comment]
+1 for correct quotes.

**Comment by swisspol** [comment]
There has been no activity for some time on this PR so I'm closing it so it's removed from the active PR list. Feel free to re-open.

**Pull request closed without merging**